devtools/forcefield-scripts/amber/conversion_script.py

```diff
@@ -0,0 +1,148 @@
+    system_omm = parm_omm.createSystem(splitDihedrals=True)
+    ala3_omm_energies = parmed.openmm.energy_decomposition_system(parm_omm, system_omm, nrg=kilojoules_per_mole)
+    # Test villin headpiece
+    # AMBER
+    print('Calculating villin headpiece energies...')
+    parm_amber = parmed.load_file(villin_top[1], villin_crd[1])
+    system_amber = parm_amber.createSystem(splitDihedrals=True)
+    villin_amber_energies = parmed.openmm.energy_decomposition_system(parm_amber, system_amber, nrg=kilojoules_per_mole)
+    # OpenMM
+    ff = app.ForceField(ffxml_name)
+    system_omm = ff.createSystem(parm_amber.topology)
+    parm_omm = parmed.openmm.load_topology(parm_amber.topology, system_omm, xyz=parm_amber.positions)
+    system_omm = parm_omm.createSystem(splitDihedrals=True)
+    villin_omm_energies = parmed.openmm.energy_decomposition_system(parm_omm, system_omm, nrg=kilojoules_per_mole)
+
+    print('Deleting temp files...')
+    for f in (ala3_top, ala3_crd, villin_top, villin_crd, leap_script_ala3_file,
+             leap_script_villin_file):
+        os.unlink(f[1])
+
+    print('Asserting ala_ala_ala energies...')
+    counter = 0
+    for i, j in zip(ala3_amber_energies, ala3_omm_energies):
+        if counter != 3: # Not Impropers
+            testing.assert_allclose(j[1], i[1], rtol=1e-5,
+            err_msg=('Ala_ala_ala energies outside of allowed tolerance for %s' % ffxml_name))
+            counter += 1
+        else: # Impropers - higher tolerance
+            testing.assert_allclose(j[1], i[1], rtol=1e-2,
+            err_msg=('Ala_ala_ala energies outside of allowed tolerance for %s' % ffxml_name))
+            counter += 1
+    print('Ala_ala_ala energy validation successful!')
+
+    print('Assering villin headpiece energies...')
+    counter = 0
+    for i, j in zip(villin_amber_energies, villin_omm_energies):
+        if counter != 3: # Not Impropers
+            testing.assert_allclose(j[1], i[1], rtol=1e-5,
+            err_msg=('Villin energies outside of allowed tolerance for %s' % ffxml_name))
+            counter += 1
+        else: # Impropers - higher tolerance
+            testing.assert_allclose(j[1], i[1], rtol=1e-2,
+            err_msg=('Villin energies outside of allowed tolerance for %s' % ffxml_name))
+            counter += 1
+    print('Villin headpiece energy validation successful!')
+    print('Done!')
+
+if __name__ == '__main__':
+    if os.getenv('AMBERHOME'):
+        AMBERHOME = os.getenv('AMBERHOME')
+    else:
+        raise RuntimeError('$AMBERHOME not set')
+    if not os.path.exists('ffxml/'):
+        os.mkdir('ffxml')
+    data = yaml.load(open('files/master.yaml'))
+    ignore = {'solvents.lib', 'atomic_ions.lib', 'ions94.lib', 'ions91.lib'}
+    for entry in data:
+        leaprc_name = entry['Source']
+        leaprc_reference = entry['Reference']
+        print('Converting %s to ffxml...' % leaprc_name)
+        filename = os.path.join(AMBERHOME, 'dat/leap/cmd', leaprc_name)
```

Comment:
Instead of having `$AMBERHOME` set, if we have `ambermini` installed, can we just use something like
```python
from distutils.spawn import find_executable
AMBERHOME = None
try:
    AMBERHOME = os.environ['AMBERHOME']
except KeyError:
    pass

if AMBERHOME is None:
    full_path = find_executable("parmchk2")
    try:
        AMBERHOME = os.path.split(full_path)[0]
        AMBERHOME = os.path.join(AMBERHOME, "../")
    except:
        raise(ValueError("Cannot determine AMBERHOME automatically"))

if AMBERHOME is None:
    raise(ValueError("Cannot determine AMBERHOME. Set $AMBERHOME or install ambermini"))
```